tests/test_black.py

```diff
@@ -1104,6 +1104,15 @@ def test_cache_single_file_already_cached(self) -> None:
             with src.open("r") as fobj:
                 self.assertEqual(fobj.read(), "print('hello')")
 
+    def test_cache_read_from_broken_pickle(self) -> None:
+        mode = black.FileMode()
+        from shutil import copyfile
+
+        src = THIS_DIR / "data" / "cache-38.pickle"
```

Comment:
Let me guess this based on file name:
* this is legit py3.8 pickle file
* test is assumed to be ran on py3.7

Would it be better to cook up a pickle file with some crazy protocol number instead?

Reply:
Correct. And yes; well; it's currently run on 3.6 and 3.7 during CI checks.

But you're right that an improbable pickle protocol number would be better, since this test could become obsolete. I didn't get at all into pickle internals to set this up (just pickled a tiny module in 3.8).